stratosphere/ncm/source/lr_registeredlocationresolver.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2019 Adubbz
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "lr_registeredlocationresolver.hpp"
+
+namespace sts::lr {
+
+    RegisteredLocationResolverInterface::RegisteredLocationResolverInterface() {
+        this->registered_program_redirector.ClearRedirections();
+        this->registered_html_docs_redirector.ClearRedirections();
+    }
+
+    RegisteredLocationResolverInterface::~RegisteredLocationResolverInterface() {
+        /* Ensure entries are deallocated */
+        this->html_docs_redirector.ClearRedirections();
+        this->program_redirector.ClearRedirections();
+    }
+
+    Result RegisteredLocationResolverInterface::ResolveProgramPath(OutPointerWithServerSize<Path, 0x1> out, ncm::TitleId tid) {
+        Path path;
+        
+        if (!this->program_redirector.FindRedirection(&path, tid)) {
```

Comment:
this should definitely be a helper that takes in a redirector + a registeredlocations and tries them in order.